packages/ember-data/lib/serializers/json-serializer.js

```diff
@@ -1459,6 +1460,7 @@ var JSONSerializer = Serializer.extend({
     @return {Object} json The deserialized payload
   */
   extractSingle: function(store, typeClass, payload, id, requestType) {
+    Ember.deprecate('`serializer.normalizePayload` has been deprecated. Please use `serializer.normalizeResponse` with the new Serializer API to modify the payload.', this.normalizePayload === JSONSerializer.prototype.normalizePayload);
```

Comment:
I put the deprecation here instead of inside the `normalizePayload` method since overriding would lose the deprecation. We're only displaying it if the user has overridden `normalizePayload`:

```js
'...modify the payload.', this.normalizePayload === JSONSerializer.prototype.normalizePayload);
```